docs/guide/creating-variadics/README.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Creating Variadic Patch Nodes
+---
+
+# Creating Variadic Patch Nodes
+
+Declaring your patch node to be variadic allows its user to add or remove
+similar input pins dynamically. For you, it is a way to avoid creating a list
+of mechanical variations to support the various number of inputs like:
+
+* `foobar`
+* `foobar-2`
+* `foobar-3`
+* …
+* `foobar-8`
+
+Make sure you’re familiar with the [variadic nodes](../variadics/) concept
+and understand how the expansion process works.
+
+## Markers
+
+To make your patch node variadic, you should place a special marker node on it.
+Find the marker node under the `xod/patch-nodes` library. You need
+`variadic-1`, `variadic-2`, or `variadic-3`. The number defines the arity step,
+that is the group size by which the rightmost input pins replicate.
+
+The `variadic-*` node itself does nothing and is stripped out during
+compilation. It exists solely to define indefinite arity support.
+
+## Pin assignments
+
+Whenever you place a variadic node marker, the input pins get roles depending
+on their position relative to each other. Consider a patch node draft with a
+`variadic-2` marker:
+
+![Pin assignments](./assignments.patch.png)
+
+To make the expansion XOD split pins of a variadic patch node into three
+groups.
+
+**Variadic pins** (aka value pins). The rightmost pins get this role. Their
+number equals to the arity step. In the example above they are `V1` and
+`V2`. Values for the inputs on each level are provided by a node user.
+
+**Accumulator pins**. Pins in the middle get this role. Their number equals the
+output number. In the example they are `A1`, `A2`, and `A3`. The user provides
+values only for the top level of the cascade; all other levels get accumulator
+values straightly from output pins of the respective upstream node.
+
+**Shared pins**. The leftmost pins which are not variadic nor accumulator get
+this role. In the example they are `S1` and `S2`. The values are provided by
+the user, and they are delivered to each level in the cascade as is. In many
+cases there would be null shared pins.
```

Comment:
"null"?

Reply:
:thinking: zero? nil?

Reply:
IMHO, "zero" is fine. Or even "there would be **no** shared pins"